### PR TITLE
implements step for adding updateinfo.xml into a repository

### PR DESCRIPTION
**Pull request by kkaarreell**
Implements step 
  Given updateinfo defined in repository "{repository}"
that let's you describe the content of updateinfo.xml file which is then
added to an existing (baseurl=file://) repository.

### CODE REVIEW

File: dnf-docker-test/features/steps/updateinfo_steps.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    modifyrepo = which("modifyrepo_c")
+    ctx.assertion.assertIsNotNone(modifyrepo, "modifyrepo_c is required")
+
+    # prepare updateinfo.xml content
+    repodir = updateinfo_utils.get_repo_dir(repository)
+    updateinfo_xml = updateinfo_utils.get_updateinfo_xml(repository, updateinfo_table)
+
+    # save it to the updateinfo.xml file and recreate the repodata
+    tmpdir = tempfile.mkdtemp()
+    fw = open('%s/updateinfo.xml' % tmpdir, 'w')
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/updateinfo_steps.py]: * use context manager for this (`with open() as fw:`)
* Use `os.path.join(tmpdir, "updateinfo.xml")`

File: dnf-docker-test/features/steps/updateinfo_steps.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import table_utils
+import updateinfo_utils
+from whichcraft import which
+
+from behave import given
+from command_steps import step_i_successfully_run_command
+
+
+@given('updateinfo defined in repository "{repository}"')
+def given_updateinfo_defined_in_repository(ctx, repository):
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/updateinfo_steps.py]: naming is `step_`, not `given_` ;)

File: dnf-docker-test/features/steps/updateinfo_steps.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    Requires table with following headers:
+
+    ==== ===== =======
+     Id   Tag   Value
+    ==== ===== =======
+
+    *Tag* is describing attributes of the respective update.
+    Supported tags are:
+
+    ============== ====================
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/updateinfo_steps.py]: Would be nice to have width same as length of biggest line

File: dnf-docker-test/features/steps/updateinfo_steps.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    Description    Default description of Id
+    Summary        Default summary of Id
+    Severity       Low
+    Solution       Default solution of Id
+    Rights         Chuck Norris
+    Issued         2017-01-01 00:00:01
+    Updated        2017-01-01 00:00:01
+    Reference      none
+    Package        none
+    ============== ===============
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/updateinfo_steps.py]: should have length same as heading

File: dnf-docker-test/features/steps/updateinfo_utils.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        cnt = doc.createTextNode(pkg_details[rpm]['pkg_sum'])
+        elem.appendChild(cnt)
+        package.appendChild(elem)
+        # add package into collection
+        collection.appendChild(package)
+    return elem_update
+
+
+def get_updateinfo_xml(repository, updateinfo_table):
+    primary_xml = get_repodata_file_path('.*primary.xml.gz$', repository)
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/updateinfo_utils.py]: don't really like hardcoding of `.gz`, probably should be something like `.*primary\.xml.*`?

File: dnf-docker-test/features/steps/updateinfo_utils.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        d['pkg_sum'] = pkg.getElementsByTagName("checksum")[0].firstChild.data
+        d['pkg_sumtype'] = pkg.getElementsByTagName("checksum")[0].getAttribute("type")
+        pkg_dict[d['pkg_filename']] = d
+    return pkg_dict
+
+
+def build_updateinfo_xml_elem_update(update, pkg_details):
+    doc = xml.dom.minidom.Document()
+    elem_update = doc.createElement("update")
+    elem_update.setAttribute("from", "security@redhat.com")
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/updateinfo_utils.py]: I would use `nobody@dnf.baseurl.org`...

File: dnf-docker-test/features/steps/updateinfo_utils.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+REPO_TMPL = "/etc/yum.repos.d/{!s}.repo"
+
+
+def get_repo_dir(repository):
+    repofile = REPO_TMPL.format(repository)
+    rf = open(repofile, 'r')
+    repodir = ''
+    for line in rf.readlines():
+        if line.startswith('baseurl = file://'):
+            repodir = line[17:].strip()
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/updateinfo_utils.py]: use configparser for this instead

File: dnf-docker-test/features/steps/updateinfo_utils.py (hunk 5 above)
Comment (kkaarreell) [inline dnf-docker-test/features/steps/updateinfo_utils.py]: Well, I need to open a file I am able to uncompress. Currently there would be *.sqlite and *.gz files, we can add the support for other formats later but currently *.gz seems to be the best choice.

File: dnf-docker-test/features/steps/updateinfo_steps.py (hunk 2 above)
Comment (kkaarreell) [inline dnf-docker-test/features/steps/updateinfo_steps.py]: I have changed it but fyi, this naming is used in other files too.

**Pull request closed without merging**